source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -192,6 +193,14 @@ namespace PlayFab
                     {
                         // Add HTTP headers
                         SetPredefinedHeaders(reqContainer, hRequest);
+                        if(!setPredefinedHeadersFailed)
+                        {
```

Comment:
note this if-expression also reads backwards. `if (!failed) { /* failure case */ }`

you should be checking `if(setPredefinedHeadersFailed) { /* failure case */ }` but SetPredefinedHeaders is returning a backwards result